### DISCUSSION TITLE
Spam comments on a livestream chat

### DISCUSSION THREAD

**Discussion by RickyRocky** [ThioJoe/YT-Spammer-Purge#339] [Ideas]
On the chat of Youtube livestreaming there can be spam comments. Will the program detect them?

An example of spam comments is the spam that the vtuber Shirakami Fubuki sometimes gets. When she collabs in other people channels spam comments can be found.

In this video, https://youtu.be/uO8a7-GZ7l4?t=250 (@4:10), there is spam in the chat.
![chat_spam](https://user-images.githubusercontent.com/97205567/148321915-8e4657e8-b571-4693-91de-aa98b66fa286.JPG)
Right now must of it is like that. 4 to 5 nouns repeated with a username that could be a legal name.

This video, https://youtu.be/CziZPfafgCU?t=253 (@4:13), is another example of the spam comments.
![chat_spam2](https://user-images.githubusercontent.com/97205567/148322287-eb3ae675-20d9-477e-af1f-0f0952964ba6.JPG)

I don't know if the spammers still use it but earlier this year they used this program https://github.com/zhenshiluosuo/Storyteller-AutoBarrageForDouyuTV
Maybe this could be useful to understand how it works.

**Comment by RickyRocky** [comment]
Another example of the spam in the chat.
https://youtu.be/Z2IwcJxNk-c?t=4413 (@1:13:33)
![chat1](https://user-images.githubusercontent.com/97205567/148353013-c8880373-fe8c-4d36-aeef-87a1c279218f.JPG)
In this kind the spam is a text in Japanese that is spammed by accounts with usernames with non-Japanese legal names.

https://youtu.be/26s3Ubj1jwE?t=6667(@1:51:07)
![chat2](https://user-images.githubusercontent.com/97205567/148354111-96f3a714-2fb8-4e66-ac21-1aea05f78416.JPG)

https://youtu.be/-v9EMFBQdYc?t=6867 (@1:54:27)
![chat3](https://user-images.githubusercontent.com/97205567/148353581-5596aae5-309c-42bc-b64e-f436b994b39d.JPG)

Other streams with the spam: https://youtu.be/hX_naaso17M?t=1717 (@28:37)
https://youtu.be/VVe2B8TRlKQ?t=2270 (@37:50)
https://youtu.be/EBlYYyjs3ak?t=1001 (@16:41)

**Comment by ThioJoe** [comment]
There might be a way to do it but it would have to be implemented much differently. As of right now, it seems like when a video is live, the API marks it as "comments disabled" so you can't even scan existing comments on it until it's over. I'm not sure if there is any kind of 'chat stream' you can access

**Reply by TechStudent10** [reply]
Maybe build something like a YouTube bot that people use in chats to moderate. Not sure how, but I know it's possible with custom bots being built. For example, I was on a Kotski (YouTube Analytic Stream, Live Sub Counts, etc.) stream a while ago and they were using a custom bot.

Edit: Corrected Name

**Comment by Kurdakov** [comment]
there are commercial offerings like Nightbot 
and few github projects (did not test them though)
https://github.com/pkspyder007/youtube-live-chat-moderator
https://github.com/teremich/michis-mod-bot - this one is in python, but with previous API (2) which is deprecated (newer one is here https://developers.google.com/youtube/v3/live/getting-started )
